src/controllers/EvidenceDownloadController.ts

```diff
@@ -13,16 +18,22 @@ import { DOWNLOAD_FILE_PAGE_URI } from 'app/utils/Paths';
 const template = 'download-file';
 const errorCustomTemplate = 'error-custom';
 
-@controller(DOWNLOAD_FILE_PAGE_URI, FileTransferFeatureMiddleware)
+@controller(DOWNLOAD_FILE_PAGE_URI,
+    SessionMiddleware,
+    AuthMiddleware,
+    FileTransferFeatureMiddleware,
+    LoadAppealMiddleware,
+    FileRestrictionsMiddleware
+)
 export class EvidenceDownloadController extends BaseAsyncHttpController {
 
     constructor(@inject(FileTransferService) private readonly fileTransferService: FileTransferService) {
         super();
     }
 
     @httpGet('/prompt/:fileId')
-    public renderPrompt(@requestParam('fileId') fileId: string): void {
-        this.httpContext.response.render(template, { fileId });
+    public async renderPrompt(@requestParam('fileId') fileId: string): Promise<void> {
+        return this.render(template, { downloadPath: this.getDownloadPath(this.httpContext.request, fileId) });
     }
 
     @httpGet('/data/:fileId/download')
```

Comment:
Are the newly added parameters on the download link read by this endpoint at any point? or are they only used by the LoadAppealMiddleware?

Reply:
They are ready by the `this.getDownloadPath` function

Reply:
Ok, so what's the reason for generating a download path with the appeal ID and companyNumber if they parameters are then not read by the download endpoint? i'm assuming by this point, the appeal has already been loaded and is in the session. Or am i missing something here?

Reply:
They are used to create the download link in the prompt page

Reply:
discussed